vortex-array/src/stats/bound.rs

```diff
@@ -0,0 +1,312 @@
+use std::cmp::Ordering;
+
+use vortex_error::{VortexError, VortexResult};
+use Precision::Inexact;
+
+use crate::stats::Precision::Exact;
+use crate::stats::{Precision, StatBound};
+
+/// Interpret the value as a lower bound.
+/// These format a partial order over successively more precise bounds
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct LowerBound<T>(pub(crate) Precision<T>);
+
+impl<T> LowerBound<T> {
+    pub(crate) fn min_value(self) -> T {
+        self.0.into_value()
+    }
+}
+
+impl<T> LowerBound<T> {
+    pub fn is_exact(&self) -> bool {
```

Comment:
Can this be `as_exact(&self) -> Option<T>` instead? Probably more useful

Reply:
And seems odd that this isn't some trait function of a Bound trait. Same with min_value -> Bound::value

Reply:
These min_value -> value doesn't make sense, it is min and max value rsp.